### ISSUE TITLE
Allow both optional and required secrets from AWS SecretsManager

### ISSUE THREAD

**Issue by chrisrhut** [awspring/spring-cloud-aws#114]
**Type**: Feature

**Is your feature request related to a problem? Please describe.**
Using Spring Boot's configuration import it appears that when fetching multiple keys from the Secrets Manager, all must be optional or required, but a combination is not allowed.

This configuration fails for example:

```yaml
spring:
  config:
    import:
      - aws-secretsmanager:/secret/app/required
      - optional:aws-secretsmanager:/secret/app/optional
```
The application fails to start with the stacktrace:

```
java.lang.IllegalStateException: Could not register object [com.amazonaws.services.secretsmanager.AWSSecretsManagerClient@198386a2] under bean name 'configDataAWSSecretsManager': there is already object [com.amazonaws.services.secretsmanager.AWSSecretsManagerClient@198386a2] bound
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.registerSingleton(DefaultSingletonBeanRegistry.java:124) ~[spring-beans-5.3.5.jar:5.3.5]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.registerSingleton(DefaultListableBeanFactory.java:1138) ~[spring-beans-5.3.5.jar:5.3.5]
	at io.awspring.cloud.autoconfigure.secretsmanager.AwsSecretsManagerConfigDataLocationResolver.lambda$registerAndPromoteBean$1(AwsSecretsManagerConfigDataLocationResolver.java:113) ~[spring-cloud-starter-aws-secrets-manager-config-2.3.1.jar:2.3.1]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:176) ~[spring-context-5.3.5.jar:5.3.5]
```

**Describe the solution you'd like**
It would be great if optional and required imports could be combined. This would allow some interesting and complex configurations to be used, such as a profile with a required base configuration but an optional add-on configuration based on some environment variable (or an additional active profile).

Thank you very much for considering this feature!

**Comment by chrisrhut** [comment]
Tangentially related: I run into the same startup error if I try to "layer" configurations like so:

**application.yaml**
```yaml
spring.config.import: aws-secretsmanager:/secret/app
```

**application-p1.yaml**
```yaml
spring.config.import: aws-secretsmanager:/secret/app/p1
```

Is there a workaround to build up a configuration like this with multiple profiles?

**Comment by eddumelendez** [comment]
Hi @chrisrhut, thanks for the report! indeed both required and optional can be applied but I think the stacktrace is related to another issue. could you share a small sample to reproduce the error, please?  See https://github.com/awspring/spring-cloud-aws/issues/108

**Comment by chrisrhut** [comment]
Hi @eddumelendez, here's a minimal test that reproduces the problem. Please let me know if I can provide this in a different format... _Note,_ you'll just need to replace these two lines in the integration test file with secrets that actually exist in your environment, of course I can't provide those :sweat_smile: 

```
@SpringBootTest(properties = {
  "spring.config.import[0]=aws-secretsmanager:/secret/app/a",
  "spring.config.import[1]=optional:aws-secretsmanager:/secret/app/b",
})
```

**pom.xml:**
```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>
  <parent>
    <groupId>org.springframework.boot</groupId>
    <artifactId>spring-boot-starter-parent</artifactId>
    <version>2.4.4</version>
    <relativePath/>
  </parent>
  <groupId>example</groupId>
  <artifactId>aws-secrets-test</artifactId>
  <version>0.0.1-SNAPSHOT</version>
  <name>aws-secrets-test</name>
  <description>Demo project for Spring Boot</description>
  <properties>
    <java.version>11</java.version>
  </properties>
  <dependencies>
    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter</artifactId>
    </dependency>
    <dependency>
      <groupId>io.awspring.cloud</groupId>
      <artifactId>spring-cloud-starter-aws-secrets-manager-config</artifactId>
      <version>2.3.0</version>
    </dependency>
    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-test</artifactId>
      <scope>test</scope>
    </dependency>
  </dependencies>
  <build>
    <plugins>
      <plugin>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-maven-plugin</artifactId>
      </plugin>
    </plugins>
  </build>
</project>
```

**src/main/java/example/AwsSecretsTestApplication.java:**
```java
package example;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

@SpringBootApplication
public class AwsSecretsTestApplication {
  public static void main(String[] args) {
    SpringApplication.run(AwsSecretsTestApplication.class, args);
  }
}
```

**src/test/java/example/AwsSecretsTestApplicationTests.java:**
```java
package example;

import org.junit.jupiter.api.Test;
import org.springframework.boot.test.context.SpringBootTest;

@SpringBootTest(properties = {
  "spring.config.import[0]=aws-secretsmanager:/secret/app/a",
  "spring.config.import[1]=optional:aws-secretsmanager:/secret/app/b",
})
class AwsSecretsTestApplicationTests {
  @Test
  void contextLoads() {
  }
}
```

Running this yields the exception:

```
java.lang.IllegalStateException: Could not register object [com.amazonaws.services.secretsmanager.AWSSecretsManagerClient@596a7f44] under bean name 'configDataAWSSecretsManager': there is already object [com.amazonaws.services.secretsmanager.AWSSecretsManagerClient@596a7f44] bound
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.registerSingleton(DefaultSingletonBeanRegistry.java:124) ~[spring-beans-5.3.6.jar:5.3.6]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.registerSingleton(DefaultListableBeanFactory.java:1138) ~[spring-beans-5.3.6.jar:5.3.6]
	at io.awspring.cloud.autoconfigure.secretsmanager.AwsSecretsManagerConfigDataLocationResolver.lambda$registerAndPromoteBean$1(AwsSecretsManagerConfigDataLocationResolver.java:113) ~[spring-cloud-starter-aws-secrets-manager-config-2.3.0.jar:2.3.0]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:176) ~[spring-context-5.3.6.jar:5.3.6]
```

Thanks again!

**Comment by MatejNedic** [comment]
Hey @chrisrhut ,
Sorry for the late response. The issue you described has been fixed with #142 
Once released you will be able to do this without any issues:

`spring:
  config:
    import:
      - aws-secretsmanager:/secret/app/required
      - optional:aws-secretsmanager:/secret/app/optional`
      
I am closing this issue now but if you have any more questions feel free to write.
Fix will be applied with the release of 2.3.2.

**Comment by chrisrhut** [comment]
Thank you @MatejNedic, I'll give it a try!
